src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1055,6 +1055,89 @@ object Desugar {
+            )
+            // add declared and modified variables to initalization
+            val declaredLocalVars = declared.map(id => 
+              in.LocalVar(idName(id, info), typeD(info.typ(id), Addressability.outParameter)(src))(src)
+            )
+            val modifiedLocalVars = modified.map(id => 
+              in.LocalVar(idName(id, info), typeD(info.typ(id), Addressability.outParameter)(src))(src)
+            )
+            val blockInit = targets ++ declaredLocalVars
+            // create arguments for function call
+            val outArgs = variables.map(id => in.LocalVar(idName(id, info),typeD(info.typ(id), Addressability.outParameter)(src))(src))
+            // add function call
+            val proxy = in.FunctionProxy(nm.function("outline"++nm.fresh, info))(meta(n, info))
+            val functionCall = in.FunctionCall(targets, proxy, outArgs)(src)
+            // init declared variables
+            val declareInits = declaredLocalVars.map(lvar => in.Initialization(lvar)(src))
+            // assign return values to variables
+            val returnTargets = targets.zip (modifiedLocalVars ++ declaredLocalVars)
+            val returnAssigns = returnTargets.map{ case (t, r) => 
+              in.SingleAss(in.Assignee.Var(r), t)(src)
+            }
+            // create context for body translation
+            val outlinedContext = new FunctionContext(_ => violation("Outlined body can not contain return."))
+            // init new variables
+            val newVariablesInit = modifiedArgsSubs.map(v => in.Initialization(v)(v.info))
+            // assign in parameters to local variables
+            val argInits = modifiedArgsWithSubsAux.flatMap{
+              case (p, q) => Some(singleAss(in.Assignee.Var(q), p)(p.info))
+              case _ => None
+            }
+            // extends context
+            modifiedArgsSubs.zip(modified).foreach{case (lvar, id) =>
+              outlinedContext.addSubst(id, lvar)
+            }
+            // assign local variables to out parameters
+            val outParamAssigns = (modifiedArgsSubs ++ declaredLocalVars).zip(returns).map{case (lvar, param) =>               
+              //val variable = in.LocalVar(idName(id, info), typeD(info.typ(id), Addressability.exclusiveVariable)(src))(src)
+              in.SingleAss(in.Assignee.Var(param), lvar)(src)
+            }
+            // desugar outlined function body
+            val desugaredBlockStmts = sequence(n.body map (s => seqn(stmtD(outlinedContext)(s))))
+            val blockContent = blockV(desugaredBlockStmts)(src)
+            // create block
+            val blockBody = newVariablesInit ++ argInits ++ Vector(blockContent) ++ outParamAssigns
+            val block = in.Block(modifiedArgsSubs, blockBody)(src)
+            // create spec context
+            val specContext = new FunctionContext(_ => violation("Outlined body can not contain return."))
```

Comment:
Why do we have a second `FunctionContext`? Shouldn't this one be the same as the `outlinedContext`?